Packs/FeedAutofocus/Integrations/FeedAutofocus/FeedAutofocus.py

```diff
@@ -0,0 +1,242 @@
+
+        res = requests.request(
+            method="GET",
+            url=url,
+            verify=self.verify,
+            headers=self.headers
+        )
+        res.raise_for_status()
+        return res.text.split('\n')
+
+    def find_indicator_type(self, indicator):
+        """Infer the type of the indicator.
+
+        Args:
+            indicator(str): The indicator whose type we want to check.
+
+        Returns:
+            str. The type of the indicator.
+        """
+        if re.match(ipv4cidrRegex, indicator):
+            return FeedIndicatorType.CIDR
+
+        if re.match(ipv6cidrRegex, indicator):
+            return FeedIndicatorType.IPv6CIDR
+
+        if re.match(ipv4Regex, indicator):
+            return FeedIndicatorType.IP
+
+        if re.match(ipv6Regex, indicator):
+            return FeedIndicatorType.IPv6
+
+        elif re.match(sha256Regex, indicator):
+            return FeedIndicatorType.File
+
+        # in AutoFocus, URLs include '/' character while domains do not.
+        elif '/' in indicator:
+            return FeedIndicatorType.URL
+
+        else:
+            return FeedIndicatorType.Domain
+
+    def build_iterator(self):
+        """Builds a list of indicators.
+
+        Returns:
+            list. A list of JSON objects representing indicators fetched from a feed.
+        """
+        indicators = []
+
+        if "Daily Threat Feed" in self.indicator_feeds:
+            indicators.extend(self.http_request(feed_type="Daily Threat Feed"))
+
+        if "URL Feed" in self.indicator_feeds:
+            indicators.extend(self.http_request(feed_type="URL Feed"))
+
+        parsed_indicators = []
+
+        for indicator in indicators:
+            if indicator:
+                indicator_type = self.find_indicator_type(indicator)
+
+                parsed_indicators.append({
+                    "type": indicator_type,
+                    "value": indicator,
+                    "rawJSON": {
+                        'value': indicator,
+                        'type': indicator_type
+                    }
+                })
+
+        return parsed_indicators
+
+
+def module_test_command(client: Client, args: dict):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client: Autofocus Feed client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    indicator_feeds = client.indicator_feeds
+    if 'Daily Threat Feed' in indicator_feeds:
+        client.indicator_feeds = ['Daily Threat Feed']
+        try:
+            client.build_iterator()
+        except Exception:
+            raise Exception("Could not fetch Daily Threat Feed\n\nCheck your API key and your connection AutoFocus.")
+
+    if 'URL Feed' in indicator_feeds:
+        client.indicator_feeds = ['URL Feed']
+        try:
+            client.build_iterator()
```

Comment:
seems like really weird behaviour

Reply:
I just wanted to inform the user that his feed is turned off - this will cause the fetch to fail and it is specific for this type of feed.